config/management/src/lib.rs

```diff
@@ -65,6 +78,73 @@ impl Command {
         }
     }
 
+    pub fn validator_config(self) -> Transaction {
+        if let Command::ValidatorConfig(config) = self {
+            let mut storage = secure_storage(config.secure_backend);
+            let consensus_key = storage
+                .get_public_key(constants::CONSENSUS_KEY)
+                .expect("Unable to retrieve consensus key from storage")
+                .public_key;
+            let fullnode_network_key = storage
+                .get_public_key(constants::FULLNODE_NETWORK_KEY)
+                .expect("Unable to retrieve fullnode network key from storage")
+                .public_key;
+            let validator_network_key = storage
+                .get_public_key(constants::VALIDATOR_NETWORK_KEY)
+                .expect("Unable to retrieve validator network key from storage")
+                .public_key;
+
+            let fullnode_network_key = fullnode_network_key.to_bytes();
+            let fullnode_network_key: x25519::PublicKey = fullnode_network_key
+                .as_ref()
+                .try_into()
+                .expect("Unable to decode x25519 from fullnode_network_key");
+
+            let validator_network_key = validator_network_key.to_bytes();
+            let validator_network_key: x25519::PublicKey = validator_network_key
+                .as_ref()
+                .try_into()
+                .expect("Unable to decode x25519 from fullnode_network_key");
```

Comment:
this is a lot of boilerplate, s/fullnode/validator

Reply:
this is to keep people on their feet.

Reply:
I had to work around the migration to network address at which point this will simplify -- to some extent.